CONTRIBUTING.md

```diff
@@ -15,11 +15,36 @@ to discuss proposals and use pull requests to suggest changes. For more
 information see the [Cilium Development
 Guide](https://docs.cilium.io/en/latest/contributing/development/).
 
+## An important note about mutating cluster resources
+
+Cilium CLI must follow these general rules regarding mutating cluster resources:
+
+- Cilium CLI must not mutate resources that are not managed by the Cilium Helm
+  chart. Cilium CLI must use Cilium Helm values to configure Cilium installations.
+- The only exception to the rule above is the connectivity test command, which
+  may mutate resources within Kubernates namespaces with the name prefix specified
+  by --test-namespace flag. The connectivity test must not mutate resources that
+  affect network connectivity outside the test namespaces.
```

Comment:
Maybe it would be worth explicitly calling out that deploying any cluster-scoped CRD object as part of these tests is forbidden?